cdk/lib/frontend.ts

```diff
@@ -18,5 +34,360 @@ export class Frontend extends GuStack {
+          "arn:aws:dynamodb:*:*:table/redemption-codes-UAT",
+        ],
+      }),
+      new GuAllowPolicy(this, "StripeSetupIntentLambda", {
+        actions: ["lambda:InvokeFunction"],
+        resources: [
+          `arn:aws:lambda:eu-west-1:${this.account}:function:stripe-intent-${this.stage}`,
+        ],
+      }),
+    ];
+
+    const ec2App = new GuEc2App(this, {
+      applicationPort: 9000,
+      app: "frontend",
+      access: { scope: AccessScope.PUBLIC },
+      certificateProps: {
+        [Stage.CODE]: {
+          domainName:
+            "support.code.theguardian.com.origin.membership.guardianapis.com",
+          hostedZoneId: "Z1E4V12LQGXFEC",
+        },
+        [Stage.PROD]: {
+          domainName:
+            "support.theguardian.com.origin.membership.guardianapis.com",
+          hostedZoneId: "Z1E4V12LQGXFEC",
+        },
+      },
+      monitoringConfiguration: {
+        noMonitoring: true,
+      },
+      userData,
+      roleConfiguration: {
+        additionalPolicies: policies,
+      },
+      scaling: {
+        PROD: {
+          minimumInstances: 3,
+          maximumInstances: 6,
+        },
+        CODE: {
+          minimumInstances: 1,
+          maximumInstances: 2,
+        },
+      },
+      instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.SMALL),
+    });
+
+    // TODO: remove this tag after the migration
+    const ec2AppAsg = ec2App.autoScalingGroup;
+    Tags.of(ec2AppAsg).add("gu:riffraff:new-asg", "true");
+
+    // ---- Alarms ---- //
+
+    const alarmName = (shortDescription: string) =>
+      `URGENT 9-5 - ${this.stage} ${shortDescription}`;
+
+    const alarmDescription = (description: string) =>
+      `Impact - ${description}. Follow the process in https://docs.google.com/document/d/1_3El3cly9d7u_jPgTcRjLxmdG2e919zCLvmcFCLOYAk/edit`;
+
+    const alarmEnabledInProd = this.withStageDependentValue({
+      app,
+      variableName: "AlarmEnabledInProd",
+      stageValues: {
+        [Stage.CODE]: false,
+        [Stage.PROD]: true,
+      },
+    });
+
+    new GuAlarm(this, "NoHealthyInstancesAlarm", {
+      app,
+      alarmName: alarmName("no healthy instances for support-frontend"),
+      alarmDescription: alarmDescription(
+        "Cannot sell any subscriptions or contributions products"
+      ),
+      actionsEnabled: alarmEnabledInProd,
+      threshold: 0.5,
+      evaluationPeriods: 2,
+      comparisonOperator: ComparisonOperator.LESS_THAN_OR_EQUAL_TO_THRESHOLD,
+      metric: new Metric({
+        metricName: "HealthyHostCount",
+        namespace: "AWS/ApplicationELB",
+        dimensionsMap: {
+          LoadBalancer: ec2App.loadBalancer.loadBalancerFullName,
+          TargetGroup: ec2App.targetGroup.targetGroupFullName,
+        },
+        statistic: "Average",
+        period: Duration.seconds(60),
+      }),
+      snsTopicName: "reader-revenue-dev",
+    });
+
+    new GuAlarm(this, "ReducedHealthyInstancesAlarm", {
```

Comment:
IIUC there is some overlap between the goal of this alarm and the goal of the [`GuUnhealthyInstances`](https://guardian.github.io/cdk/classes/constructs_cloudwatch.GuUnhealthyInstancesAlarm.html) alarm that the pattern is [able to create for you](https://github.com/guardian/cdk/blob/d1de578b5691d83afa591a5b6c47cb91349ea70d/src/patterns/ec2-app/base.ts#L73). It'd be interesting to know if the latter could work for you, but I think it makes sense to retain your existing alarm for now.

Reply:
I'll create a health card for this one too. I think it'd be great if we can fallback on as many defaults as possible from the `GuEc2App` pattern. I'm not convinced we need two separate alarms (`NoHealthyInstances` and `ReducedHealthInstances`), so if we can make this one work it would save quite a lot of boilerplate